examples/neural_compressor/optical-character-recognition/run_trocr.py

```diff
@@ -0,0 +1,310 @@
+        )
+    else:
+        model_args, data_args, training_args, optim_args = parser.parse_args_into_dataclasses()
+
+    # Setup logging
+    logging.basicConfig(
+        format="%(asctime)s - %(levelname)s - %(name)s - %(message)s",
+        datefmt="%m/%d/%Y %H:%M:%S",
+        handlers=[logging.StreamHandler(sys.stdout)],
+    )
+
+    log_level = training_args.get_process_log_level()
+    logger.setLevel(log_level)
+    transformers.utils.logging.set_verbosity(log_level)
+    transformers.utils.logging.enable_default_handler()
+    transformers.utils.logging.enable_explicit_format()
+
+    # Log on each process the small summary:
+    logger.warning(
+        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu}"
+        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
+    )
+    logger.info(f"Training/evaluation parameters {training_args}")
+
+    test_file = os.path.join(data_args.datasets_dir, "gt_test.txt")
+    test_dir = os.path.join(data_args.datasets_dir, "image")
+    df = pd.read_fwf(test_file, header=None)
+    df.rename(columns={0: "file_name", 1: "text"}, inplace=True)
+    del df[2]
+    df.head()
+
+    class IAMDataset(Dataset):
+        def __init__(self, root_dir, df, processor, max_target_length=128):
+            self.root_dir = root_dir
+            self.df = df
+            self.processor = processor
+            self.max_target_length = max_target_length
+
+        def __len__(self):
+            return len(self.df)
+
+        def __getitem__(self, idx):
+            # get file name + text
+            file_name = self.df["file_name"][idx]
+            text = self.df["text"][idx]
+            # some file names end with jp instead of jpg, the two lines below fix this
+            if file_name.endswith("jp"):
+                file_name = file_name + "g"
+            # prepare image (i.e. resize + normalize)
+            image = Image.open(os.path.join(self.root_dir, file_name)).convert("RGB")
+            pixel_values = self.processor(image, return_tensors="pt").pixel_values
+            # add labels (input_ids) by encoding the text
+            labels = self.processor.tokenizer(text, padding="max_length", max_length=self.max_target_length).input_ids
+            # important: make sure that PAD tokens are ignored by the loss function
+            labels = [label if label != self.processor.tokenizer.pad_token_id else -100 for label in labels]
+
+            encoding = {"pixel_values": pixel_values.squeeze(), "labels": torch.tensor(labels)}
+            return encoding
+
+    processor = TrOCRProcessor.from_pretrained(model_args.model_name_or_path)
+    test_dataset = IAMDataset(root_dir=test_dir, df=df, processor=processor)
+
+    test_dataloader = DataLoader(test_dataset, batch_size=training_args.per_device_eval_batch_size)
+
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
The device will always be cpu in our case so I would replace it with
```suggestion
    device = torch.device("cpu")
```